### PR TITLE
Fix Retry-After header check

### PR DESCRIPTION
**Pull request by sjoubert**
This failed to be triggered because the response has all its headers as lower case

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [x] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/core/azure-core/src/http/retry_policy.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ bool GetResponseHeaderBasedDelay(RawResponse const& response, Delay& retryAfter)
     return true;
   }
 
-  if ((header = responseHeaders.find("Retry-After")) != responseHeadersEnd)
+  if ((header = responseHeaders.find("retry-after")) != responseHeadersEnd)
```
Comment (vhvb1989) [inline sdk/core/azure-core/src/http/retry_policy.cpp:34]: LGTM!  Thank you!
@antkmsft  the transport policy would call `toLower` to all headers read from the network.

**Review by vhvb1989 — APPROVED** [review]
@antkmsft if you think this is OK, please merge.

@Jinming-Hu , @katmsft  are we honoring the `retry-after` header in the StorageRetryPolicy?  I took a look and it looks to me that we don't. Is that per desing?

**Comment by vhvb1989** [comment]
@antkmsft we would also need changelog entry for this

**Review by ahsonkhan — APPROVED** [review]
Great catch! Thanks for the fix @sjoubert. Let's audit our code base like you mentioned Victor, that we don't have such issues elsewhere.

**Comment by ahsonkhan** [comment]
@sjoubert - how did you discover this issue, and I am curious to learn how it impacted your use case?

**Comment by sjoubert** [comment]
@ahsonkhan We are building our own library/API for Azure Digital Twin on top of the core/http part of this SDK. We've been hitting "429 - Too Many Request" and the retry attempts did not follow the replied header value.

**Comment by ahsonkhan** [comment]
That's awesome. Thanks for the context. Keep us posted on how it goes and please continue to file issues (or submit PR for changes if you observe other issues) :)

We may want to consider also adding a case insensitive comparer in our map, which will make it resilient to such issues going forward, similar to storage SDK's `Metadata`:
https://github.com/Azure/azure-sdk-for-cpp/blob/5e2e84cdbcf49497ac67b77f642c2c7d0f2f0278/sdk/storage/azure-storage-common/inc/azure/storage/common/storage_common.hpp#L54-L64

**Pull request merged**